lib/spack/spack/util/executable.py

```diff
@@ -62,63 +59,57 @@ def add_default_env(self, key, value):
         """
         self.default_env[key] = value
 
-    def add_default_envmod(self, envmod):
+    def add_default_envmod(self, envmod: EnvironmentModifications) -> None:
         """Set an EnvironmentModifications to use when the command is run."""
         self.default_envmod.extend(envmod)
 
     @property
-    def command(self):
-        """The command-line string.
-
-        Returns:
-            str: The executable and default arguments
-        """
+    def command(self) -> str:
+        """Returns the entire command-line string"""
         return " ".join(self.exe)
 
     @property
-    def name(self):
-        """The executable name.
-
-        Returns:
-            str: The basename of the executable
-        """
+    def name(self) -> str:
+        """Returns the executable name"""
         return PurePath(self.path).name
 
     @property
-    def path(self):
-        """The path to the executable.
-
-        Returns:
-            str: The path to the executable
-        """
+    def path(self) -> str:
+        """Returns the executable path"""
         return str(PurePath(self.exe[0]))
 
-    def __call__(self, *args, **kwargs):
-        """Run this executable in a subprocess.
+    def __call__(
+        self,
+        *args: str,
+        fail_on_error: bool = True,
+        ignore_errors: Union[int, Sequence[int]] = (),
+        ignore_quotes: Optional[bool] = None,
+        timeout: Optional[int] = None,
+        env: Optional[Union[Dict[str, str], EnvironmentModifications]] = None,
+        extra_env: Optional[Union[Dict[str, str], EnvironmentModifications]] = None,
+        input=None,
```

Comment:
This is `Optional[typing.TextIO]` and we should disallow the other values. I don't believe a string (for a filename) is actually supported so the doc should be updated

Reply:
Done in https://github.com/spack/spack/pull/48468/commits/8d58774f58cd29acfdc59daadda45a5503f416dd